gst/nnstreamer/nnstreamer_profile.c

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * @file	nnstreamer_profile.h
+ * @date	14 April 2020
+ * @brief	Internal util for profile log.
+ * @see		http://github.com/nnstreamer/nnstreamer
+ * @author	Jaeyun Jung <jy1210.jung@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#include "nnstreamer_log.h"
+#include "nnstreamer_profile.h"
+
+/**
+ * @brief Hashtable to handle the profile tasks.
+ */
+static GHashTable *g_nns_profile_task = NULL;
+
+/**
+ * @brief Struct for the profile task.
+ */
+typedef struct
+{
+  gint64 start;
+  gint64 total;
+  gsize count;
+} nns_profile_task_s;
+
+/**
+ * @brief Internal function to free profile task.
+ */
+static void
+nns_profile_destroy_task (gpointer data)
+{
+  nns_profile_task_s *task;
+
+  task = (nns_profile_task_s *) data;
+
+  /* free all resources in task */
+  g_free (task);
+}
+
+/**
+ * @brief Function to set start time.
+ */
+void
+nns_profile_start (const gchar * name)
+{
+  nns_profile_task_s *task;
+
+  if (g_nns_profile_task == NULL) {
+    g_nns_profile_task =
```

Comment:
Q: Is it okay not to consider any locking mechanism to access this global hash table?